api/v1alpha1/nonadminbackup_types.go

```diff
@@ -32,14 +32,12 @@ type NonAdminBackupSpec struct {
 
 	// BackupSpec defines the specification for a Velero backup.
 	BackupSpec *velerov1api.BackupSpec `json:"backupSpec,omitempty"`
```

Comment:
why not make this inline as well?

Reply:
That's the API design decision, we could...

Reply:
I think it's easier for us to check:
 1. If there is no BackupSpec within NonAdminBackup we can clearly say it's missing
 2. If there is BackupSpec within NonAdminBackup, we relay on the errors and statuses from original Velero Backup

If we have inline, we would have troubles distinguishing what belongs to the BackupSpec or what was added on top of this (in the future) as part of the NonAdminBackup specification and then we would need to have some extra logic for updating the status based on that inline structure. So we would end up comparing almost every single BackupSpec to ensure status is from Velero and not NonAdminBackup itself.